src/Player.ts

```
@@ -1421,6 +1424,15 @@ export class Player {
       });
     }
 
+    // Propose a new action to undo last action
+    private undoTurnOption(game: Game): PlayerInput {
+      return new SelectOption("Undo Turn", () => {
+        //this.actionsTakenThisRound = 0;
+        game.restoreLastSave();
```

If there is an error with `restoreLastSave` I would throw an error to allow the user to know we were unable to restore their last save. We should leverage transactions if possible with the database so if there is some error with our query none of it goes through.